### PR TITLE
BFD-1592 Introduce bfd-db-migrator AMI

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
[BFD-1592](https://jira.cms.gov/browse/BFD-1592)

**User Story or Bug Summary:**

Define bfd-db-migrator AMI

---

### What Does This PR Do?
This both defines the `bfd-db-migrator` Amazon Machine Image (AMI) **and** instructs the existing `Build App AMIs` stage to build it. This does not impact the subsequent deployment stages nor is this AMI used.

This PR introduces the necessary resources to provision the bfd-db-migrator AMI, including:
* extension to the application-specific Jenkinsfile for producing and storing the artifact in the Jenkins pipeline
* extension to the operations-specific Jenkinsfile for generating the specific AMI
* new packer definition and ansible provisioner playbook to configure the bfd-db-migrator AMI
* new environment-specific ansible variables for bfd-db-migrator functionality

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/ansible/playbooks-ccs/build_bfd-db-migrator.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      import_role:
+        name: cloudwatch-agent-instrumented
+
+    - name: Apply latest host security patches
+      import_role:
+        name: base_patch
+
+    - name: Apply DB Migrator Role
+      import_role:
+        name: bfd-db-migrator
```
Comment (mackec1) [inline ops/ansible/playbooks-ccs/build_bfd-db-migrator.yml:35]: Assuming we are expecting Keith's role-based stuff is a pre-requisite?

**Review by mackec1 — APPROVED** [review]
looks good.

File: ops/ansible/playbooks-ccs/build_bfd-db-migrator.yml (hunk 1 above)
Comment (mjburling) [reply ops/ansible/playbooks-ccs/build_bfd-db-migrator.yml:35]: The migrator is _effectively_ a prereq of Keith's proposed database role changes, but this _ansible role_ is orthogonal.

File: ops/packer/build_bfd-all.json [hunk 2]
```diff
@@ -58,6 +58,33 @@
         "Branch": "{{user `git_branch`}}",
         "Commit": "{{user `git_commit`}}"
       }
+    },
+    {
+      "name": "bfd-db-migrator",
+      "type": "amazon-ebs",
+      "iam_instance_profile": "bfd-packer",
+      "ssh_username": "ec2-user",
+      "instance_type": "m5.xlarge",
```
Comment (christrotteradhoc) [inline ops/packer/build_bfd-all.json:67]: non-blocking question -- do we need this size of an instance?

**Review by christrotteradhoc — APPROVED** [review]

File: ops/packer/build_bfd-all.json (hunk 2 above)
Comment (mjburling) [reply ops/packer/build_bfd-all.json:67]: To be clear, this is the instance type that's created to serve as the source of the resultant AMI by the _packer builder_ `amazon-ebs` in the following, summarized process:
1. An ec2 instance of type `m5.xlarge` is created via the `amazon-ebs` _packer builder_
2. The routines encoded in the `ansible` _packer provisioner_  run against this instance
3. An AMI is generated from the instance

More on this builder can be found [here](https://www.packer.io/plugins/builders/amazon/ebs).

**Pull request merged**